src/cuga/backend/server/main.py

```
@@ -3411,6 +3387,39 @@ def _read_file():
         raise HTTPException(status_code=500, detail=f"Failed to load file: {str(e)}")
 
 
+@app.post("/api/workspace/upload")
+async def upload_workspace_file(
+    request: Request,
+    file: UploadFile = File(...),
```

Ruff (0.15.18)

[warning] Do not perform function call `File` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)